crypto/md5/build.info

```diff
@@ -17,11 +17,15 @@ ENDIF
 $COMMON=md5_dgst.c md5_one.c md5_sha1.c $MD5ASM
 SOURCE[../../libcrypto]=$COMMON
 SOURCE[../../providers/libimplementations.a]=$COMMON
+IF[{- !$disabled{dso} -}]
```

Comment:
I see `deprecated-3.0` checks (and not `dso`) ones for, e.g., md2 and md4 (as well as comments supposedly explaining why).  I'm not nsure I understand why DSO is the right conditional here.

Reply:
A `no-deprecated` `no-shared` build ends up with double function definitions otherwise.

The issue is MD5 which is needed in the legacy provider for one of the spliced algorithms, however it resides in the default provider.  A `no-deprecated` build removes the external definition from libcrypto and this means that the code needs to be in liblegacy.  However, when building without 'dso', liblegacy is included in libcrypto.